### PR TITLE
WT-6563 Add test confirming modify correctness.

### PR DESCRIPTION
**Pull request by agorrod**
Also extend the cursor API to allow dynamic configuration of forced eviction as a debug option.

### CODE REVIEW

**Comment by agorrod** [comment]
@luke-pearson I'd appreciate it if you can take a look and let me know if this does the same thing you were testing with WT-6563. I didn't reproduce a failure, but I hooked up the release-evict functionality differently, so it's possible that the test is no longer triggering the same code path.

**Comment by agorrod** [comment]
@sulabhM I added you here because part of this change is adding a cursor interface that force-evicts pages when we are done with them, and I'd appreciate your input on whether the approach seems good.

**Review by sulabhM — APPROVED** [review]
LGTM.

**Review by luke-pearson — APPROVED** [review]
LGTM, I fixed the test and tested it against a failing commit from Aug 13 (f7050ca66777243465acafe173be70c87bac9334).

File: test/suite/test_hs19.py [hunk 1]
```diff
@@ -52,7 +78,7 @@ def test_hs19(self):
         # to reconstruct the final value, corrupting the resulting value.
         self.session.begin_transaction()
         cursor.set_key(str(0))
-        mods = [wiredtiger.Modify('AAAAAAAAAA', 102, 10)]
+        mods = [wiredtiger.Modify('AAAAAAAAAA', 102, 0)]
```
Comment (agorrod) [inline test/suite/test_hs19.py:81]: Sorry - I changed this because I didn't understand it. Could you explain what this parameter does?

Comment (luke-pearson) [reply test/suite/test_hs19.py:83]: This parameter specifies the number of bytes to be replaced by the modify. 0 results in the modify being inserted.

**Pull request merged**